internal/pkg/service/stream/sink/router/pipeline.go

```diff
@@ -0,0 +1,131 @@
+package router
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/utctime"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition/key"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/sink/pipeline"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+type pipelineRef struct {
+	*router
+	sinkKey key.SinkKey
+	// lock protects pipeline field
+	lock sync.Mutex
+	// pipeline to write data to the sink,
+	// it is initialized when the first record is received.
+	pipeline pipeline.Pipeline
+
+	openError      error
+	openBackoff    backoff.BackOff
+	openRetryAfter time.Time
+}
+
+// pipeline gets or creates sink pipeline.
+func (r *router) pipeline(ctx context.Context, timestamp time.Time, sinkKey key.SinkKey) (pipeline.Pipeline, error) {
```

Comment:
Why is router method in `pipeline` file? This should be in my opinion in `router.go`. The definition is elsewhere and first function after `type` definition is definiton of another file.

Reply:
It is related, but I can move it, if you want.

Reply:
https://github.com/keboola/keboola-as-code/pull/1906